pkg/query/flamegraph_table.go

```diff
@@ -475,3 +481,85 @@ func equalsByNameTable(tables TableGetter, a, b *querypb.FlamegraphNode) bool {
 	strings := tables.Strings()
 	return strings[aFunction.NameStringIndex] == strings[bFunction.NameStringIndex]
 }
+
+type FlamegraphChildren []*querypb.FlamegraphNode
+
+func (n FlamegraphChildren) Cumulative() int64 {
+	cumulative := int64(0)
+	for _, child := range n {
+		cumulative += child.Cumulative
+	}
+	return cumulative
+}
+
+func (n FlamegraphChildren) Diff() int64 {
+	diff := int64(0)
+	for _, child := range n {
+		diff += child.Diff
+	}
+	return diff
+}
+
+func TrimFlamegraph(ctx context.Context, tracer trace.Tracer, graph *querypb.Flamegraph, thresholdRate float32) *querypb.Flamegraph {
+	ctx, span := tracer.Start(ctx, "trimFlamegraph")
+	defer span.End()
+	if graph == nil {
+		return nil
+	}
+	total := graph.Total
+
+	threshold := int64(float64(thresholdRate) * float64(total))
+	var children FlamegraphChildren = trimFlamegraphNodes(ctx, tracer, graph.Root.Children, threshold)
+	newTotal := int64(0)
+	newDiff := int64(0)
+	if len(graph.Root.Children) > 0 {
+		newTotal = children.Cumulative()
+		newDiff = children.Diff()
+	} else {
```

Comment:
I don't understand this else statement. If there are no children in the root node it should return 0 anyway, so this else should not be needed?

Reply:
Hmm. True, I was using this for the `FlamegraphNode` type where if there is no `Children` would mean leaf node and will have to take its own value. Sounds unnecessary for the Root node, will make it `0`.